File: Week02/week02/二叉树层序遍历/levelOrder1.py
from sklearn.tree import DecisionTreeClassifier
class Solution(object):
	def levelOrder1(self,root:'root'):
		if root is None:
			return []
		result = []
		previous_layer=[root]
		while previous_layer:
			current_layer=[]
			result.append([])
			for node in previous_layer:
				result[-1].append(node.val)
				current_layer.extend(node.children)
			previous_layer=current_layer
		return result

File: Week06/Week02/week02/二叉树层序遍历/levelorder1.java
package yiyi;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;



class Node{
	public Node() {}
	public Node(int _val) {
		val=_val;
	}
	public Node(int _val,List<Node> _children) {
		val= _val;
		children=_children;
	}
	public int val;
	public List<Node> children;

	

	public List<List<Integer>> levelorder1 (Node root){
		List<List<Integer>> result= new ArrayList<>();
		if(root==null) return result;
		List<Node> previousLayer=Arrays.asList(root);
		while(!previousLayer.isEmpty()) {
			List<Node> currentLayer=new ArrayList<>();
			List<Integer> previousVals=new ArrayList<>();
			for(Node node:previousLayer){
				previousVals.add(node.val);
				currentLayer.addAll(node.children);	
			}
			result.add(previousVals);
			previousLayer=currentLayer;
		}
		return result;
	}
}